packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,759 @@
+          child: Column(
+            children: <Widget>[
+              _verticalSpacer,
+              if (widget.leading != null)
+                ...<Widget>[
+                  if (_extendedAnimation.value > 0)
+                    SizedBox(
+                      width: lerpDouble(widget.preferredWidth, widget.extendedWidth, _extendedAnimation.value),
+                      child: widget.leading,
+                    )
+                  else
+                    widget.leading,
+                  _verticalSpacer,
+                ],
+              Expanded(
+                child: Column(
+                  mainAxisAlignment: destinationsAlignment,
+                  children: <Widget>[
+                    for (int i = 0; i < widget.destinations.length; i++)
+                      _RailDestinationBox(
+                        width: widget.preferredWidth,
+                        extendedWidth: widget.extendedWidth,
+                        extendedTransitionAnimation: _extendedAnimation,
+                        selected: widget.currentIndex == i,
+                        icon: widget.currentIndex == i ? widget.destinations[i].activeIcon : widget.destinations[i].icon,
+                        label: widget.destinations[i].label,
+                        destinationAnimation: _destinationAnimations[i],
+                        labelType: labelType,
+                        iconTheme: widget.currentIndex == i ? selectedIconTheme : unselectedIconTheme,
+                        labelTextStyle: widget.currentIndex == i ? selectedLabelTextStyle : unselectedLabelTextStyle,
+                        onTap: () {
+                          widget.onDestinationSelected(i);
+                        },
+                        indexLabel: localizations.tabLabel(
+                          tabIndex: i + 1,
+                          tabCount: widget.destinations.length,
+                        ),
+                      ),
+                  ],
+                ),
+              ),
+              if (widget.trailing != null)
+                if (_extendedAnimation.value > 0)
+                  SizedBox(
+                    width: lerpDouble(widget.preferredWidth, widget.extendedWidth, _extendedAnimation.value),
+                    child: widget.trailing,
+                  )
+                else
+                  widget.trailing,
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  MainAxisAlignment _resolveGroupAlignment(NavigationRailGroupAlignment groupAlignment) {
+    switch (groupAlignment) {
+      case NavigationRailGroupAlignment.top:
+        return MainAxisAlignment.start;
+      case NavigationRailGroupAlignment.center:
+        return MainAxisAlignment.center;
+      case NavigationRailGroupAlignment.bottom:
+        return MainAxisAlignment.end;
+    }
+    return MainAxisAlignment.start;
+  }
+
+  void _disposeControllers() {
+    for (final AnimationController controller in _destinationControllers)
```

Comment:
I personally prefer using `{ }` for all control flow statements but up to you.

Reply:
Done